### PR TITLE
Added support for subject name strategy.

### PR DESCRIPTION
**Pull request by fipil**
This PR added a support for the subject name strategy. It's based on this article [https://www.confluent.io/blog/put-several-event-types-kafka-topic/]

### CODE REVIEW

File: src/Confluent.SchemaRegistry/CachedSchemaRegistryClient.cs [hunk 1]
```diff
@@ -121,6 +124,14 @@ public CachedSchemaRegistryClient(IEnumerable<KeyValuePair<string, string>> conf
                 throw new ArgumentException($"Invalid value '{basicAuthSource}' specified for property '{SchemaRegistryConfig.PropertyNames.SchemaRegistryBasicAuthCredentialsSource}'");
             }
 
+            var keySubjNameStrategy = Convert.ToString(config.FirstOrDefault(prop => prop.Key.ToLower() == SchemaRegistryConfig.PropertyNames.SchemaRegistryKeySubjectNameStrategy).Value) ?? "";
+            var valueSubjNameStrategy = Convert.ToString(config.FirstOrDefault(prop => prop.Key.ToLower() == SchemaRegistryConfig.PropertyNames.SchemaRegistryValueSubjectNameStrategy).Value) ?? "";
+
+            if(Enum.TryParse<SubjectNameStrategy>(keySubjNameStrategy, out var strategy))
```
Comment (mhowlett) [inline src/Confluent.SchemaRegistry/CachedSchemaRegistryClient.cs]: keep formatting consistent with other parts of the codebase (space after if, always use `{}` even when not necessary).

**Comment by mhowlett** [comment]
thanks! I just remembered there's also an alternate implementation in #585. That's proof of sufficient demand. Going to be hard to prioritize this until July (flat out trying to meet a deadline for something else), but shouldn't be too far after that.

File: src/Confluent.SchemaRegistry/CachedSchemaRegistryClient.cs (hunk 1 above)
Comment (fipil) [reply src/Confluent.SchemaRegistry/CachedSchemaRegistryClient.cs]: Hi, I corrected it and pushed.

**Comment by alexpedrero** [comment]
One observation: to keep exactly the same pattern used at java client, we need to take off the "-value" suffix at RecordNameStrategy and TopicRecordNameStrategy. @fipil Thanks for implementation. @mhowlett  Can this be merged?

**Comment by mhowlett** [comment]
thanks for the feedback @alexpedrero. yes, expect to get to review this v. soon. i'm finishing something else up and this is next on the list. targeting 1.3.

**Comment by mhowlett** [comment]
closing in favor of #1070 (which extends this PR) - since we're in a hurry to get this in, I reviewed and made the required edits myself. I think / hope it'll still show you as a contributor @fipil

**Comment by fipil** [comment]
Ok, that's fine. Thanks!

**Pull request closed without merging**